### PR TITLE
Add some additional capabilities to namespace resolution

### PR DESCRIPTION
**Pull request by rlburkes**
Fix proposes a change to allow `extends` to explicitly provide a `namespace` and changes default object lookup to use the current `namespace` before falling back to global search across namespaces (this is the current behavior).

### CODE REVIEW

File: spec/avro/dsl/test/external_type.rb [hunk 1]
```diff
@@ -1,2 +1,2 @@
-namespace :test
+namespace 'test'
```
Comment (tjwp) [inline spec/avro/dsl/test/external_type.rb]: Why was this change necessary? FWIW if I eliminate the `namespace` arg for `eval_file`, then this does not seem necessary.

Comment (rlburkes) [reply spec/avro/dsl/test/external_type.rb]: Yep, sorry removed now.

File: lib/avro/builder/dsl.rb [hunk 2]
```diff
@@ -117,8 +117,16 @@ def create_named_type(name, avro_type_name, options = {}, &block)
                                                          &block)
       end
 
-      def eval_file(name)
-        file_path = find_file(name)
+      def eval_file(name, namespace = nil)
+        file_path = if namespace
+                      begin
+                        find_file([namespace, name].join('.'))
+                      rescue FileNotFoundError => _
```
[the remark this answers is not in the document]
Comment (rlburkes) [reply lib/avro/builder/dsl.rb]: Removed

File: lib/avro/builder/dsl.rb [hunk 3]
```diff
@@ -50,7 +50,7 @@ def record(name = nil, options = {}, &block)
       # Imports from the file with specified name fragment.
       def import(name)
         previous_namespace = namespace
-        result = eval_file(name)
+        result = eval_file(name, previous_namespace)
```
[the remark this answers is not in the document]
Comment (rlburkes) [reply lib/avro/builder/dsl.rb]: Removed

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,9 @@
 # avro-builder changelog
 
+## v0.16.0
+- Add support for providing an explicit `namespace` on `extend`. Default lookup
```
Comment (tjwp) [inline CHANGELOG.md]: Minor, but consider splitting this as two separate bullets?

Comment (rlburkes) [reply CHANGELOG.md]: 👍
